runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -1151,8 +1154,14 @@ iree_status_t iree_hal_hip_device_add_asynchronous_cleanup(
     iree_hal_device_t* base_device, iree_hal_hip_cleanup_callback_t callback,
     void* user_data) {
   iree_hal_hip_device_t* device = iree_hal_hip_device_cast(base_device);
-  return iree_hal_hip_cleanup_thread_add_cleanup(device->cleanup_thread, NULL,
-                                                 callback, user_data);
+
+  // Perform cleanup synchronously on main thread if cleanup thread is
+  // deinitialized
```

Comment:
style: punctuation in comments (https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar)
```suggestion
  // Perform cleanup synchronously on main thread if cleanup thread is
  // deinitialized.
```